julia-repl.el

```diff
@@ -101,6 +103,126 @@ in your Emacs init file after loading this package."
+                                       name executable-path switches)
+  (let ((inferior-buffer (apply #'make-term name executable-path nil switches)))
+    (with-current-buffer inferior-buffer
+      (mapc (lambda (k)
+              (define-key term-raw-map k (global-key-binding k)))
+            julia-repl-captures)
+      (term-char-mode)
+      (when julia-repl-set-term-escape
+        (term-set-escape-char ?\C-x))      ; useful for switching windows
+      (setq-local term-prompt-regexp "^(julia|shell|help\\?|(\\d+\\|debug ))>")
+      (setq-local term-suppress-hard-newline t)  ; reflow text
+      (setq-local term-scroll-show-maximum-output t)
+      ;; do I need this?
+      (setq-local term-scroll-to-bottom-on-output t))
+    inferior-buffer))
+
+(cl-defmethod julia-repl--send-to-backend ((_terminal-backend julia-repl--buffer-ansi-term)
+                                           buffer string paste-p ret-p)
+  "Send a string to BUFFER using the given backend.
+
+When PASTE-P, “bracketed paste” mode will be used. When RET-P, terminate with an extra newline."
+  (with-current-buffer buffer
+    (when paste-p                       ; bracketed paste start
+      (term-send-raw-string "\e[200~"))
+    (term-send-raw-string string)
+    (when ret-p                         ; return
+      (term-send-raw-string "\^M"))
+    (when paste-p                       ; bracketed paste stop
+      (term-send-raw-string "\e[201~"))))
+
+;;; vterm
+
+(with-eval-after-load 'vterm
+
+  (defun julia-repl--next-error-function (n &optional reset)
+    "A workaround for https://github.com/akermu/emacs-libvterm/issues/316."
+    ;; NOTE remove when that issue is fixed
+    (interactive "p")
+    (goto-char (point))
+    (compilation-next-error-function n reset))
+
+  (cl-defstruct julia-repl--buffer-vterm
+    "Terminal backend using ‘vterm’, which needs to be installed and loaded.")
+
+  (cl-defmethod julia-repl--locate-live-buffer ((_terminal-backend julia-repl--buffer-vterm)
+                                                name)
+    (if-let ((inferior-buffer (get-buffer (julia-repl--add-earmuffs name))))
+        (with-current-buffer inferior-buffer
+          (cl-assert (eq major-mode 'vterm-mode) nil "Expected vterm-mode. Changed mode or backends?")
+          ;; cf https://github.com/akermu/emacs-libvterm/issues/270
+          (when (let ((proc (buffer-local-value 'vterm--process inferior-buffer)))
+                  (and proc (memq (process-status proc) '(run stop open listen connect))))
+            inferior-buffer))))
+
+  (cl-defmethod julia-repl--make-buffer ((_terminal-backend julia-repl--buffer-vterm)
+                                         name executable-path switches)
+    (let ((vterm-buffer (get-buffer-create (julia-repl--add-earmuffs name)))
+          (inhibit-read-only t))
+      (with-current-buffer vterm-buffer
+        (let ((vterm-shell (apply #'concat executable-path " " switches))
```

Comment:
This should be
```emacs-lisp
(let ((vterm-shell (concat executable-path " " (mapconcat #'identity switches " ")))
```
At the moment, all options are concatenated without spaces like `-Jsysimage.so--threads8`.

Reply:
Thanks! I fixed this with `s-join` since we are using that package anyway.